src/App.tsx

```diff
@@ -1,14 +1,45 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleModal, setVisibleModal] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Status>(Status.ALL);
+  const [query, setQuery] = useState('');
+
+  const visibleTodos = useMemo(() => {
+    const currentTodos = query === '' ? todos : todos.filter(todo => (
+      todo.title.toLowerCase().includes(query.toLowerCase())
+    ));
+
+    switch (filter) {
+      case Status.ALL:
+        return currentTodos;
+
+      case Status.ACTIVE:
+        return currentTodos.filter(todo => !todo.completed);
+
+      case Status.COMPLETED:
+        return currentTodos.filter(todo => todo.completed);
+
+      default:
+        return currentTodos;
+    }
+  }, [filter, query, todos]);
+
+  useEffect(() => {
+    getTodos().then(setTodos);
```

Comment:
You are not handling the error that might occur during the API call. It's a good practice to handle the potential error in a try-catch block or then-catch chain for promise-based functions.